### PR TITLE
Image fetching optimization: automatically decide between JPEG and PNG

### PR DESCRIPTION
**Pull request by sinergise-anze**
As an optimization, we can decide for some layers (which don't output transparent values where the data is available) to use JPEG instead of PNG depending on availability of data. This PR adds `MimeTypes.JPEG_OR_PNG` "format", which causes this library to search for tiles and change format to JPEG (if the whole area is covered) or PNG (if not).

Further optimization is possible: we could search for the data availability for the whole screen to avoid excessive number of requests; however, this is something that can be improved in the next PR.

### CODE REVIEW

File: src/layer/AbstractDEMLayer.ts [hunk 1]
```diff
@@ -71,13 +71,13 @@ export class AbstractDEMLayer extends AbstractSentinelHubV3Layer {
   }
 
   public async getMap(params: GetMapParams, api: ApiType, reqConfig?: RequestConfiguration): Promise<Blob> {
-    const getMapValue = await ensureTimeout(async innerReqConfig => {
+    params = await this.decideJpegOrPng(params);
```
Comment (Ardweaden) [inline src/layer/AbstractDEMLayer.ts:74]: Shouldn't this also be inside the timeout?

File: src/layer/AbstractLayer.ts [hunk 2]
```diff
@@ -201,19 +243,29 @@ export class AbstractLayer {
     bbox: BBox,
     fromTime: Date,
     toTime: Date,
-    maxFindTilesRequests: number = 50,
-    tilesPerRequest: number = 50,
+    maxFindTilesRequests: number | null = 50,
```
Comment (Ardweaden) [inline src/layer/AbstractLayer.ts:246]: Why is this needed? Isn't it already assigned a value if not passed?

File: src/layer/AbstractLayer.ts [hunk 3]
```diff
@@ -201,19 +243,29 @@ export class AbstractLayer {
...
+    tilesPerRequest: number | null = 50,
     reqConfig?: RequestConfiguration,
+    overrideOrbitTimeMinutes: number | null = null,
```
Comment (Ardweaden) [inline src/layer/AbstractLayer.ts:249]: Why is this needed? (and set to infinity)

File: src/layer/AbstractLayer.ts (hunk 2 above)
Comment (Ardweaden) [reply src/layer/AbstractLayer.ts:246]: I assume because you wanted to pass `overrideOrbitTimeMinutes` but didn't want to use `undefined`.

File: src/layer/AbstractDEMLayer.ts (hunk 1 above)
Comment (sinergise-anze) [reply src/layer/AbstractDEMLayer.ts]: Good point - yes, it should be, and it should also pass `innerReqConfig` along, to make sure timeout is observed. Thank you for pointing this out! Fixed.

File: src/layer/AbstractLayer.ts (hunk 2 above)
Comment (sinergise-anze) [reply src/layer/AbstractLayer.ts:249]: Yes - in other places this library uses `null` as marker that some parameter was not specified and should be set to the default value. I am not sure if this was the best possible decision, but I used the same convention here for consistency.

Does that sound like a good idea or do you think a different approach would be better?

Comment (Ardweaden) [reply src/layer/AbstractLayer.ts:249]: I think this is an example why it's better to have a params object than separate params.

**Review by Ardweaden — APPROVED** [review]

File: src/layer/AbstractLayer.ts (hunk 2 above)
Comment (sinergise-anze) [reply src/layer/AbstractLayer.ts:249]: Yes, but then you _must_ name all the parameters, every time. Unfortunately JS doesn't have a real "named params" mechanism like Python does.

Comment (sinergise-anze) [reply src/layer/AbstractLayer.ts:249]: When calling, you need to name the parameters:
```
  myFunction({ parameterA: 15, parameterB: 30 });
```

You can't just say:
```
  myFunction({15, 50});  // invalid syntax
```

Anyway... this is probably a discussion for the chat. :)
